sktime/forecasting/toto.py

```diff
@@ -0,0 +1,470 @@
+    """
+
+    _tags = {
+        "y_inner_mtype": ["pd.DataFrame"],
+        "X_inner_mtype": "None",
+        "scitype:y": "both",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": False,
+        "X-y-must-have-same-index": True,
+        "enforce_index_type": None,
+        "capability:missing_values": False,
+        "capability:insample": False,
+        "capability:pred_int": True,
+        "capability:pred_int:insample": False,
+        # contribution and dependency tags
+        "authors": ["JATAYU000", "Datadog"],
+        # Datadog for Datadog/toto
+        "maintainers": ["JATAYU000"],
+        "python_version": ">= 3.10",
+        "python_dependencies": ["torch>=2.5", "toto-ts>=0.1.3"],
+    }
+
+    def __init__(
+        self,
+        seed=None,
+        id_mask=None,
+        padding_mask=None,
+        num_samples: int = 1,
+        samples_per_batch: int = 1,
+        prediction_type: str = "median",
+        scale_factor_exponent: int = 10,
+        stabilize_with_global: bool = True,
+        use_memory_efficient_attention: bool = False,
+        model_path: str = "Datadog/Toto-Open-Base-1.0",
+        device=None,
+    ):
+        self.model_path = model_path
+        self.device = device
+        self.num_samples = num_samples
+        self.samples_per_batch = samples_per_batch
+        self.use_memory_efficient_attention = use_memory_efficient_attention
+        if self.use_memory_efficient_attention:
+            if _check_soft_dependencies("xformers", severity="warning"):
+                self.set_tags(python_dependencies=["torch", "xformers", "accelerate"])
+            else:
+                raise ImportError(
+                    """
+                    xformers is required for memory efficient attention.
+                    Refer to https://github.com/facebookresearch/xformers
+                    """
+                )
+        self.stabilize_with_global = stabilize_with_global
+        self.scale_factor_exponent = scale_factor_exponent
+        self.prediction_type = prediction_type
+        if prediction_type not in ["mean", "median"]:
+            raise ValueError("prediction_type must be either 'mean' or 'median'")
+
+        self.seed = seed
+        self._seed = np.random.randint(0, 2**31) if seed is None else seed
+
+        # Original Implementation
+        self.id_mask = id_mask
+        self.padding_mask = padding_mask
+
+        super().__init__()
+
+    def _get_toto_key(self):
+        """Get a unique key for the Toto model based on configuration parameters.
+
+        This key is used by the _multiton decorator to ensure only one instance
+        of a model with specific parameters exists.
+
+        Returns
+        -------
+        tuple
+            Unique identifier for this model configuration
+        """
+        kwargs = self._get_toto_kwargs()
+        key = {
+            **kwargs,
+            "device": self._device,
+            "id_mask": self.id_mask,
```

Comment:
should `id_mask` and `padding_mask` be included here? Do these really result in different instances?